src/lsm/manifest_log.zig

```diff
@@ -266,25 +239,20 @@ pub fn ManifestLogType(comptime Storage: type, comptime TableInfo: type) type {
             const block_reference = manifest_log.read_block_reference.?;
             verify_block(block, block_reference.checksum, block_reference.address);
 
-            const entry_count = Block.entry_count(block);
-            const labels_used = Block.labels_const(block)[0..entry_count];
-            const tables_used = Block.tables_const(block)[0..entry_count];
+            const block_schema = schema.Manifest.from(block);
+            const labels_used = block_schema.labels_used(block);
+            const tables_used = block_schema.tables_used(block);
 
             const manifest: *SuperBlock.Manifest = &manifest_log.superblock.manifest;
 
-            var entry = entry_count;
+            var entry = block_schema.entry_count;
             while (entry > 0) {
                 entry -= 1;
 
                 const label = labels_used[entry];
                 const table = &tables_used[entry];
 
-                if (manifest.insert_table_extent(
-                    manifest_log.tree_id,
-                    table.address,
-                    block_reference.address,
-                    entry,
-                )) {
+                if (manifest.insert_table_extent(table.address, block_reference.address, entry)) {
```

Comment:
Good idea wrt checking the table id! I'll add that.

Including `checksum`s in the extent doesn't work, though: During `ManifestLog.append()` we need to create the table extent, but we don't yet know the block checksum (since the block in question isn't constructed yet).